frontend/app-development/layout/SettingsModalButton/SettingsModal/SettingsModal.tsx

```diff
@@ -63,6 +51,26 @@ export const SettingsModal = ({ isOpen, onClose, org, app }: SettingsModalProps)
   const localChangesTabId: SettingsModalTab = 'localChanges';
   const accessControlTabId: SettingsModalTab = 'accessControl';
 
+  /**
+   * Handles the logic for when to navigate in to a tab's content instead of continuing
+   * default behaviour.
+   */
+  const handleKeyTab = (e: React.KeyboardEvent<HTMLButtonElement>) => {
+    if (e.key === 'Enter') {
+      setEnterHasBeenClicked(true);
+    }
+    // If tabbing after clicking enter, set focus to the content in the tab
+    if (e.key === 'Tab' && enterHasBeenClicked) {
+      e.preventDefault();
+      setEnterHasBeenClicked(false);
+
+      const selectedTab = document.getElementById(`tab-content-${currentTab}`);
+      if (selectedTab) {
+        selectedTab.focus();
+      }
+    }
+  };
```

Comment:
Denne bruken av `useState` er ikke helt heldig. Det gjør at komponenten rerendres uten at noe egentlig har endret seg. `hasBeenClicked` er også et noe uklart navn - det sier ikke noe om hva som skal til for at den skal gå tilbake til `false`. Og er denne informasjonen i det hele tatt nødvendig? Vi vet jo hvilken fane som er valgt.

Uansett ser det ikke ut som at dette løser problemet. I en fanebasert visning navigerer man mellom fanene ved å bruke piltastene. Den vanlige måten å løse det på er å sette [tabindex](https://www.w3schools.com/tags/att_global_tabindex.asp) til `-1` på alle knappene unntatt den som skal ha fokus, og endre dette hver gang brukeren klikker på en piltast. Da vil navigering med piltastene fungere som ventet, og `tab`-knappen vil flytte fokus til neste element etter menyen.

Denne siden forklarer dette mer i detalj og har også noen eksempler på hvordan man kan løse det med kode: https://www.w3.org/WAI/ARIA/apg/patterns/tabs/
Her er et eksempel fra designsystemet: https://github.com/digdir/designsystem/blob/main/packages/react/src/components/legacy/LegacyTabs/Tabs.tsx

Reply:
Her er også et eksempel fra vårt eget repo: https://github.com/Altinn/altinn-studio/blob/master/frontend/packages/shared/src/components/TreeView/TreeViewItem/TreeViewItem.tsx
Dette er riktignok trevisning, hvor reglene for tastaturnavigasjon er mer omfattende.

Reply:
Det er et veldig godt poeng! Dette hadde jeg ikke tenkt på i det hele tatt. 
Jeg skal refaktorere slik at det fungerer på denne måten 😄

Reply:
@TomasEng, da skal det være fikset! 😄